crates/runtime/src/accelerated_table/refresh_task.rs

```diff
@@ -182,42 +189,118 @@ impl RefreshTask {
             vec![("dataset", dataset_name.to_string())],
         );
 
+        let start_time = SystemTime::now();
+
         let get_data_update_result = match mode {
             RefreshMode::Full => self.get_full_update().await,
             RefreshMode::Append => self.get_incremental_append_update().await,
         };
 
-        let (start_time, streaming_data_update) = match get_data_update_result {
-            Ok((start_time, data_update)) => (start_time, data_update),
+        let mut streaming_data_update = match get_data_update_result {
+            Ok(data_update) => data_update,
             Err(e) => {
                 tracing::warn!("Failed to load data for dataset {dataset_name}: {e}");
                 self.mark_dataset_status(status::ComponentStatus::Error)
                     .await;
-                return Err(RetryError::transient(e));
+                return Err(e);
             }
         };
 
-        let data_update = match streaming_data_update.collect_data().await {
-            Ok(data_update) => data_update,
+        let overwrite = streaming_data_update.update_type == UpdateType::Overwrite;
+
+        let refresh_stat: Arc<RwLock<Option<RefreshStat>>> = Arc::new(RwLock::new(None));
+        let refresh_stat_ref_copy = Arc::clone(&refresh_stat);
+
+        let is_data_receiving_err: Arc<RwLock<bool>> = Arc::new(RwLock::new(false));
+        let is_data_receiving_err_ref_copy = Arc::clone(&is_data_receiving_err);
+
+        let dataset_name_copy = dataset_name.clone();
+
+        let schema = Arc::clone(&streaming_data_update.schema);
+
+        let observed_record_batch_stream = RecordBatchStreamAdapter::new(Arc::clone(&schema), {
+            stream! {
```

Comment:
Let's use stream::unfold here instead of `stream!`

Reply:
Updated, thank you!